sos/upload/targets/__init__.py

```diff
@@ -0,0 +1,778 @@
+        'upload_no_ssl_verify': False,
+        'upload_protocol': 'auto',
+        'upload_s3_endpoint': None,
+        'upload_s3_region': None,
+        'upload_s3_bucket': None,
+        'upload_s3_access_key': None,
+        'upload_s3_secret_key': None,
+        'upload_s3_object_prefix': None,
+        'upload_target': None,
+        'skip_plugins': [],
+    }
+
+    def __init__(self, parser=None, args=None, cmdline=None):
+
+        self.ui_log = logging.getLogger('sos_ui')
+        self.parser = parser
+        self.cmdline = cmdline
+        self.args = args
+
+    @classmethod
+    def name(cls):
+        """Returns the upload target's name as a string."""
+        if cls.upload_target_name:
+            return cls.upload_target_name
+        return cls.__name__.lower()
+
+    def get_commons(self):
+        return {
+            'cmdlineopts': self.hook_commons['cmdlineopts'],
+            'policy': self.hook_commons['policy'],
+            'case_id': self.hook_commons['cmdlineopts'].case_id,
+            'upload_directory': self.hook_commons['cmdlineopts']
+            .upload_directory
+        }
+
+    def set_commons(self, commons):
+        """Set common host data for the Upload targets
+            to reference
+        """
+        self.commons = commons
+
+    def pre_work(self, hook_commons):
+
+        self.hook_commons = hook_commons
+        self.commons = self.get_commons()
+        cmdline_opts = self.commons['cmdlineopts']
+        policy = self.commons['policy']
+        caseid = cmdline_opts.case_id if cmdline_opts.case_id else ""
+
+        if cmdline_opts.low_priority:
+            policy._configure_low_priority()
+
+        # Set the cmdline settings to the class attrs that are referenced later
+        # The target default '_' prefixed versions of these are untouched to
+        # allow fallback
+        self.upload_url = cmdline_opts.upload_url
+        self.upload_user = cmdline_opts.upload_user
+        self.upload_directory = cmdline_opts.upload_directory
+        self.upload_password = cmdline_opts.upload_pass
+        self.upload_archive_name = ''
+
+        self.upload_s3_endpoint = cmdline_opts.upload_s3_endpoint
+        self.upload_s3_region = cmdline_opts.upload_s3_region
+        self.upload_s3_access_key = cmdline_opts.upload_s3_access_key
+        self.upload_s3_bucket = cmdline_opts.upload_s3_bucket
+        self.upload_s3_object_prefix = cmdline_opts.upload_s3_object_prefix
+        self.upload_s3_secret_key = cmdline_opts.upload_s3_secret_key
+
+        # set or query for case id
+        if not cmdline_opts.batch and not \
+                cmdline_opts.quiet:
+            if caseid:
+                self.commons['cmdlineopts'].case_id = caseid
+            else:
+                self.commons['cmdlineopts'].case_id = input(
+                    _("Optionally, please enter the case id that you are "
+                      "generating this report for [%s]: ") % caseid
+                )
```

Comment:
If this option is really only valid for RH, is it worth just targeting there. Otherwise it would be pointless comment/statement for any other distro?

Reply:
Ditto here, `sos/policies/distros/__init__.py` has the same check like here, again worth extending, again can be done now or as extra PR.

Reply:
I moved this to the RH target now